sympy/ntheory/factor_.py

```diff
@@ -1419,6 +1419,20 @@ def divisor_count(n, modulus=1):
     return Mul(*[v + 1 for k, v in factorint(n).items() if k > 1])
 
 
+def proper_divisors(n, generator=False):
+    if not generator:
+        return divisors(n)[1:-1]
+    else:
+        x = divisors(n, True)
+        x.remove(1)
```

Comment:
Does a generator have a remove method? I suspect you want to do something like

```
for i in divisors(n, generator=True):
    if i > 1 and i < n:
        yield i
```

Reply:
How about this?
```python
x = list(divisors(n, True))
```

Reply:
That defeats the purpose of making it a generator. 

1 is always the first thing returned by divisors, so you just need to call `next(x)` to get rid of it. That doesn't get rid of the last item, though, so I think you will need a separate function as @smichr suggests (it needs to be separate so that `generator=True` works). 

Unless there is something in itertools or sympy.utilities that could do this for you. I thought maybe [islice](https://docs.python.org/3/library/itertools.html#itertools.islice) but it doesn't let you use a negative slice for the stop.

Reply:
I think the code by @smichr is quite perfect.